aea/aea_builder.py

```diff
@@ -1282,7 +1282,8 @@ def _find_import_order(
                 ),
             )
 
-            dependency_to_supported_dependencies[skill_id] = set()
+            if skill_id not in dependency_to_supported_dependencies:
+                dependency_to_supported_dependencies[skill_id] = set()
```

Comment:
@DavidMinarsch for the specific issue you showed me:

Say we got `skill_ids = [carpark_detection, generic_seller, ...]`

First iteration: we process `carpark_detection`. and we find that `generic_seller` supports `carpark_detection` (or in other words, `carpark_detection` depends on `generic_seller`)

Second iteration: we process `generic_seller` and... well, we used to reset the "supported" skills. Hence, at the end of the for loop, `carpark_detection` doesn't depend on anything, hence the problem.

Reply:
the reason why sometimes worked is because `skill_ids` is a set, and the iterator order is not always the same across different runs.

Reply:
Nice explanation. Thanks a lot and I think this solves this issue indeed. Let's see the run complete  :)